astroid/exceptions.py

```diff
@@ -256,6 +256,19 @@ class InferenceOverwriteError(AstroidError):
     """
 
 
+class ParentMissingError(AstroidError):
+    """Raised when a node which is expected to have a parent attribute is missing one
+
+    Standard attributes:
+        target: The node for which the parent lookup failed.
+    """
+
+    target = None
+
+    def __init__(self, message="Parent not found on {target!r}.", **kws):
```

Comment:
Does `message` need to be an argument? Should be enough to define it in the `__init__` body.

--
Furthermore, typing `**kws` is quite difficult. `target` is the only expected (and required) argument. It can also be assigned to `self.target` in the function body instead of using `setattr` from `AstroidError`.

Reply:
Something like this?
I did not actually set self.target. Passing the formatted message to the `super().__init__` call seemed sufficient, right?

Reply:
It got set here: https://github.com/PyCQA/astroid/blob/2d3f8ed45e7091df7968ab10d0de366431d6a00b/astroid/exceptions.py#L63-L67

I would recommend to assign `self.target` as it would make debugging a bit easier.